### PR TITLE
Experimental end to end MLIR execution

### PR DESCRIPTION
**Pull request by flaub**

### CODE REVIEW

File: plaidml2/exec/ffi.cc [hunk 1]
```diff
@@ -63,15 +82,288 @@ class PlatformAllocator : public Allocator {
   std::string device_id_;
 };
```
Comment (jbruestle) [inline plaidml2/exec/ffi.cc]: I feel like some of this code belongs somewhere other than ffi.cc

Comment (flaub) [reply plaidml2/exec/ffi.cc]: Agreed. I’m currently working on a refactor and a way to configure passes based on the target. Was considering making a new dialect for pass configuration but after some experiments I’m currently thinking of sticking with protobuf.

**Comment by flaub** [comment]
To run, use _edsl rules along with PLAIDML_EE=1 env var

**Comment by flaub** [comment]
Try:
```
PLAIDML_EE=1 bazelisk run //plaidml2/edsl:cc_mlir_test -- --gtest_filter=CppEdsl.Dot
```

**Pull request merged**